File: src/web/routes/metrics.ts
/**
 * Metrics Endpoint Handler
 *
 * Exposes Prometheus-compatible metrics for monitoring application health,
 * performance, and behavior.
 */

import { collectMetrics } from "../../lib/metrics";

/**
 * Handle GET /metrics request
 * Returns Prometheus text format metrics
 */
export function handleMetrics(): Response {
  const metrics = collectMetrics();

  return new Response(metrics, {
    status: 200,
    headers: {
      "Content-Type": "text/plain; version=0.0.4; charset=utf-8",
    },
  });
}

File: src/web/handlers/api/metrics.go
package api

import (
	"net/http"

	"github.com/edrobertsrayne/janitarr/src/metrics"
)

// MetricsHandlers handles Prometheus metrics endpoint
type MetricsHandlers struct {
	metrics *metrics.Metrics
}

// NewMetricsHandlers creates a new MetricsHandlers instance
func NewMetricsHandlers(m *metrics.Metrics) *MetricsHandlers {
	return &MetricsHandlers{
		metrics: m,
	}
}

// GetMetrics returns metrics in Prometheus text format
// GET /metrics
func (h *MetricsHandlers) GetMetrics(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "text/plain; version=0.0.4; charset=utf-8")
	w.WriteHeader(http.StatusOK)
	w.Write([]byte(h.metrics.Format()))
}
